solutions/standard/main.tf

```diff
@@ -84,7 +84,7 @@ module "cos_instance" {
 
 locals {
   cos_instance_crn = var.existing_cos_instance_crn == null ? module.cos_instance[0].cos_instance_crn : var.existing_cos_instance_crn
-  cos_kms_key_crn  = var.existing_cos_kms_key_crn != null ? var.existing_cos_kms_key_crn : module.kms[0].keys[format("%s.%s", local.kms_key_ring_name, local.kms_key_name)].crn
+  cos_kms_key_crn  = var.enable_cos_kms_encryption ? (var.existing_cos_kms_key_crn != null ? var.existing_cos_kms_key_crn : module.kms[0].keys[format("%s.%s", local.kms_key_ring_name, local.kms_key_name)].crn) : true
```

Comment:
Valid case - If `var.enable_cos_kms_encryption` is `true`, the value of `cos_kms_key_crn` depends on the nested ternary logic.

But If `var.enable_cos_kms_encryption` is `false`, then the value of `cos_kms_key_crn` is set to a boolean i.e. `true` 

I think this condition needs to be revisited once as the `cos_kms_key_crn` is `string` type and is assigned a `bool` value, when using :

```
cos_kms_key_crn               = local.cos_kms_key_crn
```

<img width="1066" alt="image" src="https://github.com/user-attachments/assets/8a6605fc-eb28-4b3e-a09c-2c415498e64f" />

Reply:
If var.enable_cos_kms_encryption is false, then key crn wont be used. So I did not get any error.
Changed it to null instead of true.